setup/native-script.ps1

```diff
@@ -1,30 +1,76 @@
+# NOTE: The scripts requires at least a version 4.0 .NET framework installed
+# To run it inside a COMMAND PROMPT against the production branch (only one supported with self-elevation) use
+# @powershell -NoProfile -ExecutionPolicy Bypass -Command "iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/NativeScript/nativescript-cli/production/setup/native-script.ps1'))"
+# To run it inside a WINDOWS POWERSHELL console against the production branch (only one supported with self-elevation) use
+# start-process -FilePath PowerShell.exe -NoNewWindow -Wait -ArgumentList "-NoProfile -ExecutionPolicy Bypass -Command iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/NativeScript/nativescript-cli/production/setup/native-script.ps1'))"
+
+# Check if latest .NET framework installed is at least 4
+$dotNetVersions = Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\NET Framework Setup\NDP' -recurse | Get-ItemProperty -name Version,Release -EA 0 | Where { $_.PSChildName -match '^(?!S)\p{L}'} | Select Version
+$latestDotNetVersion = $dotNetVersions.GetEnumerator() | Sort-Object Version | Select-Object -Last 1
+$latestDotNetMajorNumber = $latestDotNetVersion.Version.Split(".")[0]
+if ($latestDotNetMajorNumber -lt 4) {
+	Write-Host -BackgroundColor Black -ForegroundColor Red "Running this script requires at least a version 4.0 .NET framework installed"
+	if ((Read-Host "Do you want to open Microsoft's Download Center in your default browser(y/n)") -eq 'y') {
+		Start-Process -FilePath "https://www.microsoft.com/en-us/download/search.aspx?q=.net%20framework&p=0&r=10&t=&s=Relevancy~Descending"
+	}
+
+	exit 1
+}
 
 # Self-elevate
 $isElevated = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]"Administrator")
 if (-not $isElevated) {
-    start-process -FilePath PowerShell.exe -NoProfile -ExecutionPolicy Bypass -Verb Runas -Wait -Command "iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/NativeScript/nativescript-cli/production/setup/native-script.ps1'))"
-    exit 0
+	start-process -FilePath PowerShell.exe -Verb Runas -Wait -ArgumentList "-NoProfile -ExecutionPolicy Bypass -Command iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/NativeScript/nativescript-cli/production/setup/native-script.ps1'))"
+	exit 0
+}
+
+# Help with installing other dependencies
+$script:answer = ""
+function Install($programName, $message, $script, $shouldExit) {
+	if ($script:answer -ne "all") {
+		Write-Host -BackgroundColor Black -ForegroundColor Green "Allow the script to install $($programName)?"
+		Write-Host -BackgroundColor Black "Tip: Note that if you type all you won't be prompted for subsequent installations"
+		do {
+			$script:answer = (Read-Host "y/n/all").ToLower()
```

Comment:
I'd write the text like `(Y)es/(N)o/(A)ll` or something similar to both highlight the hotkey and convey its meaning.